### PR TITLE
1130 enhance logs and supportability

### PR DESCRIPTION
**Pull request by moticless**
This PR Enhance logs and supportability. This includes 
- Review printouts
- Adding enumeration to error and info messages
- Removing redundant infra for logging.

*** Yvonne: please review text at file: pkg/logging/logger.go  ***

### CODE REVIEW

File: pkg/authenticator/authenticator.go [hunk 1]
```diff
@@ -134,16 +135,17 @@ func (auth *Authenticator) Login() error {
...
 
-		return err
+		return log.PrintAndReturnError(log.CAKC014E, err, true)
 	}
 
 	certDERBlock, certPEMBlock := pem.Decode(certPEMBlock)
 	cert, err := x509.ParseCertificate(certDERBlock.Bytes)
 	if err != nil {
-		return err
+		log.ErrorLogger.Printf(log.CAKC015E, err.Error())
```
Comment (nessiLahav) [inline pkg/authenticator/authenticator.go]: why we are printing this message twice ?

File: pkg/authenticator/authenticator.go [hunk 2]
```diff
@@ -245,16 +246,14 @@ func (auth *Authenticator) ParseAuthenticationResponse(response []byte) error {
 		return err
```
Comment (nessiLahav) [inline pkg/authenticator/authenticator.go]: change it to log.PrintAndReturnError

**Comment by nessiLahav** [comment]
change all return err to use PrintAndReturnError() function instead

File: pkg/logging/logger.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+"fmt"
+"log"
+"os"
+)
+
+var InfoLogger = log.New(os.Stdout, "INFO: ", log.LUTC|log.Ldate|log.Ltime|log.Lshortfile)
+var ErrorLogger = log.New(os.Stderr, "ERROR: ", log.LUTC|log.Ldate|log.Ltime|log.Lshortfile)
+
+
+func PrintAndReturnError(errorMessage string, err error, chainErrorToMessage bool) error {
```
Comment (nessiLahav) [inline pkg/logging/logger.go]: as we spoke please try not to use this confused boolean "chainErrorToMessage"

File: pkg/logging/logger.go [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+		return fmt.Errorf("%s, reason: %s", errorMessage, err.Error())
+	}
+}
+
+// ERROR MESSAGES
+const CAKC001E string = "CAKC001E Error creating secret handler object"
+const CAKC002E string = "CAKC002E Error creating access token object"
+const CAKC003E string = "CAKC003E Error creating secrets config"
+const CAKC004E string = "CAKC004E Error creating access token object"
+const CAKC005E string = "CAKC005E Store type %s is invalid"
```
Comment (nessiLahav) [inline pkg/logging/logger.go]: align all %s to be with '%s'

File: pkg/logging/logger.go [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+var ErrorLogger = log.New(os.Stderr, "ERROR: ", log.LUTC|log.Ldate|log.Ltime|log.Lshortfile)
+
+
+func PrintAndReturnError(errorMessage string, args ...interface{}) error {
+	ErrorLogger.Output(2, fmt.Sprintf(errorMessage, args...))
+	return fmt.Errorf(fmt.Sprintf(errorMessage, args...))
+}
+
+// ERROR MESSAGES
+const CAKC001E string = "CAKC001E Error creating secret handler object"
```
Comment (yserota) [inline pkg/logging/logger.go]:
```suggestion
const CAKC001E string = "CAKC001E Error creating the secret handler object"
```

File: pkg/logging/logger.go [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+const CAKC002E string = "CAKC002E Error creating access token object"
```
Comment (yserota) [inline pkg/logging/logger.go]:
```suggestion
const CAKC002E string = "CAKC002E Error creating the access token object"
```

File: pkg/logging/logger.go [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+const CAKC003E string = "CAKC003E Error creating secrets config"
```
Comment (yserota) [inline pkg/logging/logger.go]:
```suggestion
const CAKC003E string = "CAKC003E Error creating the secrets config file"
```

File: pkg/logging/logger.go [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+const CAKC004E string = "CAKC004E Error creating access token object"
```
Comment (yserota) [inline pkg/logging/logger.go]:
```suggestion
const CAKC004E string = "CAKC004E Error creating an access token object"
```

File: pkg/logging/logger.go [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+const CAKC005E string = "CAKC005E Store type %s is invalid"
+const CAKC006E string = "CAKC006E Error deleting access token"
```
Comment (yserota) [inline pkg/logging/logger.go]:
```suggestion
const CAKC006E string = "CAKC006E Error deleting the access token"
```

File: pkg/logging/logger.go [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+const CAKC007E string = "CAKC007E Error writing access token, reason: failed to write file. Reason: %s"
+const CAKC008E string = "CAKC008E Error writing access token, reason: failed to create directory. Reason: %s"
+const CAKC009E string = "CAKC009E Error writing access token, reason: data is empty"
+const CAKC010E string = "CAKC010E Error reading access token, reason: data is empty"
+const CAKC011E string = "CAKC011E At least one of CONJUR_SSL_CERTIFICATE and CONJUR_CERT_FILE must be provided"
```
Comment (yserota) [inline pkg/logging/logger.go]:
```suggestion
const CAKC011E string = "CAKC011E A value for  either CONJUR_SSL_CERTIFICATE and CONJUR_CERT_FILE must be provided"
```

File: pkg/logging/logger.go [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+const CAKC012E string = "CAKC012E Namespace or podname can't be empty namespace=%v podname=%v"
+const CAKC013E string = "CAKC013E Client certificate not found at %s"
+const CAKC014E string = "CAKC014E Failed reading client certificate file: %s"
```
Comment (yserota) [inline pkg/logging/logger.go]:
```suggestion
const CAKC014E string = "CAKC014E Failed to read client certificate file: %s"
```

File: pkg/logging/logger.go [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+const CAKC018E string = "CAKC018E Failed to load Conjur config"
+const CAKC019E string = "CAKC019E Failed to create Conjur client from token. Reason: %s"
+const CAKC020E string = "CAKC020E Error creating Conjur secrets provider"
+const CAKC021E string = "CAKC021E Error retrieving Conjur secrets. Reason: %s"
+const CAKC022E string = "CAKC022E Failed to create k8s secrets handler"
+const CAKC023E string = "CAKC023E Failure retrieving k8s secretsHandlerK8sUseCase"
+const CAKC024E string = "CAKC024E Failure retrieving access token"
+const CAKC025E string = "CAKC025E Error parsing Conjur variable ids"
+const CAKC026E string = "CAKC026E Error retrieving Conjur k8sSecretsHandler"
+const CAKC027E string = "CAKC027E Failure updating K8s K8sSecretsHandler map"
```
Comment (yserota) [inline pkg/logging/logger.go]:
```suggestion
const CAKC027E string = "CAKC027E Failed to update  K8s K8sSecretsHandler map"
```

File: pkg/logging/logger.go [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
+const CAKC015E string = "CAKC015E Failed parsing certificate: %s. Reason: %s"
+const CAKC016E string = "CAKC016E Login failed"
+const CAKC017E string = "CAKC017E Environment variable %s must be provided"
+const CAKC018E string = "CAKC018E Failed to load Conjur config"
+const CAKC019E string = "CAKC019E Failed to create Conjur client from token. Reason: %s"
+const CAKC020E string = "CAKC020E Error creating Conjur secrets provider"
+const CAKC021E string = "CAKC021E Error retrieving Conjur secrets. Reason: %s"
+const CAKC022E string = "CAKC022E Failed to create k8s secrets handler"
+const CAKC023E string = "CAKC023E Failure retrieving k8s secretsHandlerK8sUseCase"
+const CAKC024E string = "CAKC024E Failure retrieving access token"
```
Comment (yserota) [inline pkg/logging/logger.go]:
```suggestion
const CAKC024E string = "CAKC024E Failed to retrieve access token"
```

File: pkg/logging/logger.go [hunk 14]
```diff
@@ -0,0 +1,106 @@
...
+const CAKC025E string = "CAKC025E Error parsing Conjur variable ids"
+const CAKC026E string = "CAKC026E Error retrieving Conjur k8sSecretsHandler"
+const CAKC027E string = "CAKC027E Failure updating K8s K8sSecretsHandler map"
+const CAKC028E string = "CAKC028E Failure patching K8s K8sSecretsHandler"
```
Comment (yserota) [inline pkg/logging/logger.go]:
```suggestion
const CAKC028E string = "CAKC028E Failed to patch K8s K8sSecretsHandler"
```

File: pkg/logging/logger.go [hunk 15]
```diff
@@ -0,0 +1,106 @@
...
+const CAKC044E string = "CAKC044E Failed to parse CONJUR_TOKEN_TIMEOUT. Reason: %s"
+const CAKC045E string = "CAKC045E Failed to instantiate authenticator configuration"
+const CAKC046E string = "CAKC046E Failed to instantiate storage configuration"
+const CAKC047E string = "CAKC047E Setting SECRETS_DESTINATION environment variable to 'k8s_secrets' must run as init container"
+const CAKC048E string = "CAKC048E Failed to instantiate storage handler"
+const CAKC049E string = "CAKC049E Failed to instantiate authenticator object"
+const CAKC050E string = "CAKC050E Failure authenticating"
+const CAKC051E string = "CAKC051E Failure parsing authentication response"
+const CAKC052E string = "CAKC052E Failed to handle secrets"
+const CAKC053E string = "CAKC053E Backoff exhausted"
```
Comment (yserota) [inline pkg/logging/logger.go]: Not sure what a backoff is?  Backup?

File: pkg/logging/logger.go [hunk 16]
```diff
@@ -0,0 +1,106 @@
...
+const CAKC054E string = "CAKC054E Failed to delete access token"
+const CAKC055E string = "CAKC055E Failed read SSL Certificate. Reason: %s"
```
Comment (yserota) [inline pkg/logging/logger.go]:
```suggestion
const CAKC055E string = "CAKC055E Failed to read SSL Certificate. Reason: %s"
```

File: pkg/logging/logger.go (hunk 15 above)
Comment (moticless) [reply pkg/logging/logger.go]: A term from computer networks. Renamed to :
     CAKC053E Retransmission backoff exhausted

File: pkg/logging/logger.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (moticless) [reply pkg/logging/logger.go]: The first error that we get is:  "CAKC042E Provided incorrect value for environment variable %s"

So i guess we are clear enough.

File: pkg/logging/logger.go [hunk 17]
```diff
@@ -0,0 +1,105 @@
...
+const CAKC008E string = "CAKC008E Error writing access token, reason: failed to create directory"
+const CAKC009E string = "CAKC009E Error writing access token, reason: data is empty"
+const CAKC010E string = "CAKC010E Error reading access token, reason: data is empty"
+const CAKC011E string = "CAKC011E At least one of CONJUR_SSL_CERTIFICATE and CONJUR_CERT_FILE must be provided"
+const CAKC012E string = "CAKC012E Namespace or podname can't be empty namespace=%v podname=%v"
+const CAKC013E string = "CAKC013E Client certificate not found at %s"
+const CAKC014E string = "CAKC014E Failed reading client certificate file: %s"
+const CAKC015E string = "CAKC015E Failed parsing certificate: %s"
+const CAKC016E string = "CAKC016E Login failed"
+const CAKC017E string = "CAKC017E Environment variable %s must be provided"
```
[the remark this answers is not in the document]
Comment (moticless) [reply pkg/logging/logger.go]: Not sure what you mean. We cannot find the environment variable, such as MY_POD_NAMESPACE - and we print the error:
CAKC017E Environment variable 'MY_POD_NAMESPACE' must be provided

File: pkg/logging/logger.go [hunk 18]
```diff
@@ -0,0 +1,105 @@
...
+const CAKC018E string = "CAKC018E Failed to load Conjur config"
+const CAKC019E string = "CAKC019E Failed to create Conjur client from token"
+const CAKC020E string = "CAKC020E Error creating Conjur secrets provider"
+const CAKC021E string = "CAKC021E Error retrieving Conjur secrets"
+const CAKC022E string = "CAKC022E Failed to create k8s secrets handler"
+const CAKC023E string = "CAKC023E Failure retrieving k8s secretsHandlerK8sUseCase"
+const CAKC024E string = "CAKC024E Failure retrieving access token"
+const CAKC025E string = "CAKC025E Error parsing Conjur variable ids"
```
[the remark this answers is not in the document]
Comment (moticless) [reply pkg/logging/logger.go]: done

**Pull request closed without merging**